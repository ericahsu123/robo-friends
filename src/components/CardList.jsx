import React from 'react';
import { connect } from 'react-redux';

import Card from './Card';
import filterArrayOfObjects from './utils/filterArrayOfObjects';

const CardList = (props) => {
  const searchRobots = filterArrayOfObjects(props.robots, 'name', props.query);

  return (
    <div className="ui cards">
      {searchRobots.map((robot) => {
        return <Card key={robot.id} name={robot.name} email={robot.email} />;
      })}
    </div>
  );
};

const mapStateToProps = (state) => {
  return { robots: state.robots.robots, query: state.query };
};

export default connect(mapStateToProps)(CardList);
