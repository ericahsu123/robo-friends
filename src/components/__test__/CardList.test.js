import React from 'react';
// We're using our own custom render function and not RTL's render
// our custom utils also re-export everything from RTL
// so we can import fireEvent and screen here as well
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import CardList from '../Cardlist';

afterEach(cleanup);

test('<CardList/> component matches snapshot', () => {
  const mockStore = configureStore([]);
  const store = mockStore({
    robots: {
      robots: [
        { id: 1, name: 'Aby', email: 'aby@gmail.com' },
        { id: 2, name: 'Abt', email: 'abt@mail.com' },
        { id: 3, name: 'Tse', email: 'tse@gmail.com' },
      ],
    },
    query: 'ab',
  });
  const component = renderer.create(
    <Provider store={store}>
      <CardList />
    </Provider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
