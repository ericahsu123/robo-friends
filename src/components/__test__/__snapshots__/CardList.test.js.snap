// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList/> component matches snapshot 1`] = `
<div
  className="ui cards"
>
  <div
    className="ui centered card"
  >
    <div
      className="image"
    >
      <img
        alt="robot"
        src="https://robohash.org/Aby.png?bgset=bg1 "
        style={
          Object {
            "backgroundColor": "rgb(185, 226, 224)",
          }
        }
      />
    </div>
    <div
      className="content"
    >
      <div
        className="center aligned header"
      >
        Aby
      </div>
      <div
        className="center aligned description"
      >
        aby@gmail.com
      </div>
      <span
        className="right floated"
      >
        <i
          className="star outline like icon"
          data-testid="add-count"
          onClick={[Function]}
        />
        <span
          data-testid="likes-count"
        >
          0
        </span>
        <span>
           
        </span>
        <span>
          stars
        </span>
      </span>
    </div>
  </div>
  <div
    className="ui centered card"
  >
    <div
      className="image"
    >
      <img
        alt="robot"
        src="https://robohash.org/Abt.png?bgset=bg1 "
        style={
          Object {
            "backgroundColor": "rgb(185, 226, 224)",
          }
        }
      />
    </div>
    <div
      className="content"
    >
      <div
        className="center aligned header"
      >
        Abt
      </div>
      <div
        className="center aligned description"
      >
        abt@mail.com
      </div>
      <span
        className="right floated"
      >
        <i
          className="star outline like icon"
          data-testid="add-count"
          onClick={[Function]}
        />
        <span
          data-testid="likes-count"
        >
          0
        </span>
        <span>
           
        </span>
        <span>
          stars
        </span>
      </span>
    </div>
  </div>
</div>
`;
